oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/forms/components/EDTFDatePickerField/hooks.js

```diff
@@ -0,0 +1,44 @@
+import { useState, useEffect } from "react";
+import { registerLocale } from "react-datepicker";
+
+export const useInitialDateEdtfFormat = (fieldValue) => {
+  let dateEdtfFormat;
+  if (fieldValue) {
+    const value = fieldValue.includes("/")
+      ? fieldValue.split("/")[0]
+      : fieldValue;
+    if (value.length === 4) {
+      dateEdtfFormat = "yyyy";
+    } else if (value.length === 7) {
+      dateEdtfFormat = "yyyy-mm";
+    } else {
+      dateEdtfFormat = "yyyy-mm-dd";
+    }
```

Comment:
This spliting of input and comparing it to expected lenghts feels a little brittle to me. Have you considered using the: https://www.npmjs.com/package/edtf to parse & validate the edtf input? It that library is not feasible to use, it must be ensured here, that input is always sanitized (stripped of any potential surrounding whitespace) and handle invalid input

Reply:
@mirekys Ahoj, I have implemented the library to parse the dates. Just to clarify, this hook is only in charge of returning the type of edtf (Y, YM, or YMD), in case the field is already filled with some data coming from the backend, so that the calendar would start with this date format. I have not taken into account that someone wants to fill for example 2015/2015-12 and this is currently not possible in our datepicker, which is why I just split the date in case it is a range and use the first element as a reference. If we have to support being able to choose i.e. just a year as starting date and then year month and date as ending date, then the component will need some reworking, as there is just one state that holds the edtf format and it is shared between the pickers.

Reply:
@mirekys i've also introduced some bug, so please hold off on merging

Reply:
@mirekys I think it should be OK now. Please take a look.